src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentivesSynchronisationService.kt

```diff
@@ -20,35 +21,74 @@ class IncentivesSynchronisationService(
 
   suspend fun synchroniseIncentive(iepEvent: IncentiveUpsertedOffenderEvent) {
     nomisApiService.getIncentive(iepEvent.bookingId, iepEvent.iepSeq).run {
+
       // todo integration and remove this log
       log.debug("received nomis incentive: $this")
 
-      mappingService.findNomisIncentiveMapping(iepEvent.bookingId, iepEvent.iepSeq)?.run {
+      mappingService.findNomisIncentiveMapping(
+        nomisBookingId = bookingId,
+        nomisIncentiveSequence = incentiveSequence
+      )?.let { incentiveMapping ->
         // todo remove this log
```

Comment:
switching from run to allow naming of "it" as it was getting confusing - more refactoring to make this clearer to come